### PR TITLE
DRILL-4192: Fix selection root when there is only a single top level …

### PR DESCRIPTION
**Pull request by mehant**
…partition

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java [hunk 1]
```diff
@@ -201,7 +202,7 @@ public static FileSelection create(final DrillFileSystem fs, final String parent
    *
    * @see FileSelection#FileSelection(List, List, String)
    */
-  public static FileSelection create(final List<FileStatus> statuses, final List<String> files, final String root) {
+  public static FileSelection create(final List<FileStatus> statuses, final List<String> files, String root) {
```
Comment (hnfgns) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java]: I would recommend avoiding the re-assignment to root.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java [hunk 2]
```diff
@@ -213,14 +214,15 @@ public static FileSelection create(final List<FileStatus> statuses, final List<S
...
       selectionRoot = commonPathForFiles(files);
     } else {
-      if (statuses.size() == 1 && !Strings.isNullOrEmpty(root)) {
-        final Path rootPath = new Path(root);
-        final URI uri = statuses.get(0).getPath().toUri();
-        final Path path = new Path(uri.getScheme(), uri.getAuthority(), rootPath.toUri().getPath());
-        selectionRoot = path.toString();
-      } else {
-        selectionRoot = commonPath(statuses);
+      assert !Strings.isNullOrEmpty(root);
```
Comment (hnfgns) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java]: What is the expected behavior when root is null and assertions are disabled? This throws an NPE, I think we should not.

File: exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java [hunk 3]
```diff
@@ -18,12 +18,15 @@
...
 import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.common.util.TestTools;
 import org.apache.drill.exec.planner.physical.PlannerSettings;
 import org.junit.Ignore;
 import org.junit.Test;
 
 public class TestBugFixes extends BaseTestQuery {
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(TestBugFixes.class);
+  static final String WORKING_PATH = TestTools.getWorkingPath();
+  static final String TEST_RES_PATH = WORKING_PATH + "/src/test/resources";
```
Comment (hnfgns) [inline exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java]: You may like to use TestTools.getTestResourcesPath() instead, less fail-prone.

File: exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java [hunk 4]
```diff
@@ -144,4 +147,15 @@ public void testDRILL2361_JoinColumnAliasWithDots() throws Exception {
             .build().run();
   }
 
+  @Test
+  public void testDRILL4192() throws Exception {
+    String query = (String.format("select dir0, dir1 from dfs_test.`%s/bugs/DRILL-4192` order by dir1", TEST_RES_PATH));
```
Comment (hnfgns) [inline exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java]: I am wondering if this test should also include an HDFS wildcard.

**Comment by mehant** [comment]
@hnfgns updated PR based on your review comments.

**Pull request merged**